internal/cmdsearch/searchreplace_test.go

```diff
@@ -99,6 +99,50 @@ metadata:
   name: nginx-service
 foo:
   bar: 4
+ `,
+	},
+	{
+		name: "search replace by value to different type",
+		args: []string{"--by-value", "3", "--put-literal", "four"},
```

Comment:
Do we have a way where we will actually change the type of the yaml node? This seems like a situation where we either should change the type to `string`, or if we have openapi information, make sure we chose the appropriate type for the schema.

Reply:
I discovered an easy way to change yaml type. Just unset it so that encoder will interpret the provided value. Updated the PR and added comments for yaml documentation.

Reply:
Yeah, this delegates deciding the type to the underlying yaml library. Eventually we want to leverage the openapi schema more, but this is a good solution for now.